workflow/rules/common.smk

```diff
@@ -255,14 +262,34 @@ def all_input(wildcards):
                 "results/plots/ma/{model}.ma-plots.pdf",
                 "results/plots/qq/{model}.qq-plots.pdf",
                 "results/tables/diffexp/{model}.transcripts.diffexp.tsv",
-                "results/plots/diffexp-heatmap/{model}.diffexp-heatmap.pdf",
                 "results/tables/logcount-matrix/{model}.logcount-matrix.tsv",
                 "results/sleuth/{model}.samples.tsv",
                 "results/datavzrd-reports/diffexp-{model}",
             ],
             model=config["diffexp"]["models"],
         )
     )
+    if config["diffexp"]["genes_of_interest"]["activate"] == True:
+        wanted_input.extend(
+            expand(
+                [
+                    "results/plots/diffexp-heatmap/{model}.diffexp-heatmap.{mode}.pdf",
+                ],
+                model=config["diffexp"]["models"],
+                mode=["topn", "predefined"],
+            )
+        )
+    else:
+        wanted_input.extend(
+            expand(
+                [
+                    "results/plots/diffexp-heatmap/{model}.diffexp-heatmap.{mode}.pdf",
+                ],
+                model=config["diffexp"]["models"],
+                mode=["topn"],
+            )
+        )
+        return wanted_input
```

Comment:
Why this early return?

Reply:
This way, the targets below won't be collected.

Reply:
Sorry, That was a typo mistake. Removed it.